core/serialization/blocks.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Handles serializing blocks to plain JavaScript objects only
+ *     containing state.
+ */
+'use strict';
+
+goog.module('Blockly.serialization.blocks');
+goog.module.declareLegacyNamespace();
+
+goog.require('Blockly.Xml');
+goog.require('Blockly.inputTypes');
+
+
+// TODO: Remove this once lint is fixed.
+/* eslint-disable no-use-before-define */
+
+// IMPORTANT: Use train-case for all property names to ensure compatibility
+//     with caseless formats.
+
+/**
+ * Represents the state of a given block.
+ * @typedef {{
+ *     type: string,
+ *     id: string,
+ *     x: ?number,
+ *     y: ?number,
+ *     collapsed: ?boolean,
+ *     disabled: ?boolean,
+ *     editable: ?boolean,
+ *     deletable: ?boolean,
+ *     movable: ?boolean,
+ *     inline: ?boolean,
+ *     data: ?string,
+ *     extra-state: *,
+ *     fields: ?Object<string, *>,
+ *     inputs: ?Object<string, {shadow: ?State, block: ?State}>
+ *     next: [shadow: ?State, block: ?State]
+ * }}
+ */
+// eslint-disable-next-line no-unused-vars
+var State;
+
+/**
+ * Returns the state of the given block as a plain JavaScript object.
+ * @param {!Blockly.Block} block The block to serialize.
+ * @param {{addCoordinates: (boolean|undefined), addInputBlocks:
+ *     (boolean|undefined), addNextBlocks: (boolean|undefined)}=} param1
+ *     addCoordinates: If true, the coordinates of the block are added to the
+ *       serialized state. False by default.
+ *     addinputBlocks: If true, children of the block which are connected to
+ *       inputs will be serialized. True by default.
+ *     addNextBlocks: If true, children of the block which are connected to the
+ *       block's next connection (if it exists) will be serialized.
+ *       False by default.
+ * @return {?Blockly.serialization.blocks.State} The serialized state of the
+ *     block, or null if the block could not be serialied (eg it was an
+ *     insertion marker).
+ */
+const save = function(
+    block,
+    {
+      addCoordinates = false,
+      addInputBlocks = true,
+      addNextBlocks = false
```

Comment:
It may be better to set this to true by default. Not sure.

Reply:
Originally it was because I thought there were a lot of internal things in Blockly that used this behavior. But it turns out it's just copy paste, and everything else wants next blocks. I'll switch it over to true by default.